overlord/ifacestate/apparmorprompting/common/common.go

```diff
@@ -0,0 +1,466 @@
+	Permissions []string `json:"permissions,omitempty"`
+}
+
+// ValidateForInterface returns nil if the constraints are valid for the given
+// interface, otherwise returns an error.
+func (c *Constraints) ValidateForInterface(iface string) error {
+	switch iface {
+	case "home", "camera":
+		// TODO: change to this once PR #13730 is merged:
+		// if err := ValidatePathPattern(c.PathPattern); err != nil {
+		//	return err
+		// }
+	default:
+		return fmt.Errorf("constraints incompatible with the given interface: %q", iface)
+	}
+	permissions, err := AbstractPermissionsFromList(iface, c.Permissions)
+	if err != nil {
+		return err
+	}
+	c.Permissions = permissions
+	return nil
+}
+
+// Match returns true if the constraints match the given path, otherwise false.
+//
+// If the constraints or path are invalid, returns an error.
+func (c *Constraints) Match(path string) (bool, error) {
+	// TODO: change to this once PR #13730 is merged:
+	// return PathPatternMatch(c.PathPattern, path)
+	return true, nil
+}
+
+// RemovePermission removes every instance of the given permission from the
+// permissions list associated with the constraints. If the permission does
+// not exist in the list, returns ErrPermissionNotInList.
+func (c *Constraints) RemovePermission(permission string) error {
+	origLen := len(c.Permissions)
+	for i := 0; i < len(c.Permissions); {
+		perm := c.Permissions[i]
+		if perm != permission {
+			i++
+			continue
+		}
+		copy(c.Permissions[i:], c.Permissions[i+1:])
+		c.Permissions = c.Permissions[:len(c.Permissions)-1]
+	}
+	if origLen == len(c.Permissions) {
+		return ErrPermissionNotInList
+	}
+	return nil
+}
+
+// ContainPermissions returns true if the constraints include every one of the
+// givne permissions.
+func (c *Constraints) ContainPermissions(permissions []string) bool {
+	for _, perm := range permissions {
+		if !strutil.ListContains(c.Permissions, perm) {
+			return false
+		}
+	}
+	return true
+}
+
+type OutcomeType string
+
+const (
+	OutcomeUnset OutcomeType = ""
+	OutcomeAllow OutcomeType = "allow"
+	OutcomeDeny  OutcomeType = "deny"
+)
+
+// AsBool returns the outcome as a boolean, or an error if it cannot be parsed.
+func (outcome OutcomeType) AsBool() (bool, error) {
+	switch outcome {
+	case OutcomeAllow:
+		return true, nil
+	case OutcomeDeny:
+		return false, nil
+	default:
+		return false, fmt.Errorf(`invalid outcome: must be %q or %q: %q`, OutcomeAllow, OutcomeDeny, outcome)
+	}
+}
+
+type LifespanType string
+
+const (
+	LifespanUnset    LifespanType = ""
+	LifespanForever  LifespanType = "forever"
+	LifespanSession  LifespanType = "session"
+	LifespanSingle   LifespanType = "single"
+	LifespanTimespan LifespanType = "timespan"
+)
+
+// TimestampToTime converts the given timestamp string to a time.Time in Local
+// time. The timestamp string is expected to be of the format time.RFC3339Nano.
+// If it cannot be parsed as such, returns an error.
+func TimestampToTime(timestamp string) (time.Time, error) {
```

Comment:
feels like this belong to timeutil as it doesn't have anything to prompting?  same for a few of the helpers here?